### PR TITLE
SuperMethodCodeLens - do not show stale lenses

### PR DESCRIPTION
**Pull request by dos65**
Previosuly, lenses wasn't erased if document was changed in buffer only.

### CODE REVIEW

File: tests/unit/src/test/scala/tests/SuperMethodLspSuite.scala [hunk 1]
```diff
@@ -223,6 +228,77 @@ class SuperMethodLspSuite extends BaseLspSuite("gotosupermethod") {
...
+            |""".stripMargin
+      }
+      code2 <- server.codeLensDirect("a/src/main/scala/a/A.scala")
+      _ = assertNoDiff(
+        code2,
+        """|package a
+           |trait X {
+           |  def foo: Int
+           |}
+           |//case class Y(foo: Int) extends X
```
Comment (dos65) [inline tests/unit/src/test/scala/tests/SuperMethodLspSuite.scala:290]: Before this change `<< foo>>`  was returned for this commented code

File: tests/unit/src/main/scala/tests/TestingServer.scala [hunk 2]
```diff
@@ -841,6 +841,20 @@ final class TestingServer(
     } yield TextEdits.applyEdits(textContents(filename), textEdits.toList)
   }
 
+  def codeLensDirect(filename: String): Future[String] = {
```
Comment (tgodzik) [inline tests/unit/src/main/scala/tests/TestingServer.scala:844]: Why not use the existing `codeLenses` method ?

**Review by tgodzik — APPROVED** [review]
Looks good! Just one comment

File: tests/unit/src/main/scala/tests/TestingServer.scala (hunk 2 above)
Comment (dos65) [reply tests/unit/src/main/scala/tests/TestingServer.scala:844]: hmm, now I can return back to existing one
At the begging I wanted to test that there no lenses returned which leads to failure if existing one is used. But then changed test data.

Comment (tgodzik) [reply tests/unit/src/main/scala/tests/TestingServer.scala:844]: that would be cool, since that class has already way more methods than needed.

**Review by tgodzik — APPROVED** [review]
LGTM!

**Pull request merged**